examples/map.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey"
+)
+
+// the questions to ask
+var simpleQs = []*survey.Question{
+	{
+		Name: "name",
+		Prompt: &survey.Input{
+			Message: "What is your name?",
+		},
+		Validate: survey.Required,
+	},
+	{
+		Name: "color",
+		Prompt: &survey.Select{
+			Message: "Choose a color:",
+			Options: []string{"red", "blue", "green"},
+		},
+		Validate: survey.Required,
+	},
+}
+
+func main() {
```

Comment:
Yea tbh, this makes me really nervous. I'm also not sure what you mean by structs only allowing string fields - there are many examples of non-string fields in the codebase.  

I think we need to explore what the intended use case is and try to figure out what the simplest way for the user to specify that is. I think i'm much more in favor of adding `When` to `survey.Question` which would allow for this sort of dynamic logic withou a lot of complication.